test/test_base.py

```diff
@@ -0,0 +1,86 @@
+import dataclasses
+
+import numpy as np
+
+from hippopt import (
+    OptimizationObject,
+    Parameter,
+    StorageType,
+    TOptimizationObject,
+    Variable,
+    default_storage_type,
+)
+
+
+@dataclasses.dataclass
+class TestVariable(OptimizationObject):
+    storage: StorageType = default_storage_type(Variable)
+
+
+@dataclasses.dataclass
+class TestParameter(OptimizationObject):
+    storage: StorageType = default_storage_type(Parameter)
+
+
+def test_zero_variable():
+    test_var = TestVariable()
+    test_var.storage = np.ones(shape=3)
```

Comment:
This is what I mean with the default value after construction, the zeroing logic is maybe more appropriate to be defined in the `TestVariable.__post_init__` method.

Reply:
I see! I will add the ``__post_init__`` method in the test!

Reply:
Done in https://github.com/ami-iit/hippopt/pull/3/commits/cc780037ef274e9e8d95f28218eb2d5a1221a697